kiva/basecore2d.py

```diff
@@ -1129,17 +1129,16 @@ def draw_path(self, mode=FILL_STROKE):
             for func, args in subpath:
                 if func == POINT:
                     self.draw_subpath(mode)
-                    self.add_point_to_subpath(args)
+                    self.add_point_to_subpath(args.reshape(1, 2))
```

Comment:
Is it guaranteed that `args` will always be a NumPy array? I see some examples that are calling `line_set` with lists of tuples, for example here: https://github.com/enthought/enable/blob/a71a93821ef0c15e4e5d5b47a2afbdb7db2bad9b/docs/source/kiva_tutorial/tutorial_advanced.py#L226-L228

(Same question for `LINE` below.)

Reply:
I guess calling the `np.reshape` function instead of the method might be a way to fix this.

Reply:
The values for `args` are populated from `move_to`, `line_to`, etc. which should be populating with arrays, but you're right it looks like there might be some places where this may not be absolutely guaranteed (eg. `lines` and `line_set` docstring specifies Nx2 arrays, but probably gets used with lists of points).

Possibly can move the reshaping there as well.

Reply:
I fixed the one case I could see where `lines` might add non-array object to the sub-path args, plus a second issue where closing the path could error or give the wrong results.